lib/routes/zxcs/novel.ts

```diff
@@ -0,0 +1,112 @@
+import { load } from 'cheerio';
+import cache from '@/utils/cache';
+import ofetch from '@/utils/ofetch';
+import { parseDate } from '@/utils/parse-date';
+import type { Route, DataItem } from '@/types';
+
+export const route: Route = {
+    path: '/novel/:type',
+    name: '小说列表',
+    url: 'zxcs.info',
+    maintainers: ['liaocl'],
+    example: '/zxcs/novel/jinqigengxin',
+    parameters: { type: '小说类型, 可在对应类型页 URL 中找到' },
+    description: `支持小说类型：
+    - jinqigengxin 近期更新
+    - dushi 都市
+    - xianxia 仙侠
+    - xuanhuan 玄幻
+    - qihuan 奇幻
+    - lishi 历史
+    - youxi 游戏
+    - wuxia 武侠
+    - kehuan 科幻
+    - tiyu 体育
+    - lingyi 灵异
+    - junshi 军事
+    - erciyuan 轻小说
+    `,
+    categories: ['reading'],
+    features: {
+        requireConfig: false,
+        requirePuppeteer: false,
+        antiCrawler: false,
+        supportBT: false,
+        supportPodcast: false,
+        supportScihub: false,
+    },
+    radar: [
+        {
+            source: ['zxcs.info/:type'],
+            target: '/novel/:type',
+        },
+    ],
+    handler,
+};
+
+const types = {
+    jinqigengxin: '近期更新',
+    dushi: '都市',
+    xianxia: '仙侠',
+    xuanhuan: '玄幻',
+    qihuan: ' 奇幻',
+    lishi: '历史',
+    youxi: '游戏',
+    wuxia: '武侠',
+    kehuan: '科幻',
+    tiyu: '体育',
+    lingyi: '灵异',
+    junshi: '军事',
+    erciyuan: '轻小说',
+};
+
+async function handler(ctx) {
+    const { type } = ctx.req.param();
+
+    const link = `https://www.zxcs.info/${type}`;
+    const response = await ofetch(link);
+    const $ = load(response);
+
+    const list: DataItem[] = $('div.book-info')
+        .toArray()
+        .map((item) => {
+            const $item = $(item);
+            const a = $item.find('a').first();
+            return {
+                title: a.text(),
+                link: String(a.attr('href')),
+                pubDate: parseDate($item.find('.update').text()),
+                category: [],
+                description: '',
+                image: '',
+                author: '',
+            };
+        });
+
+    const items = await Promise.all(
+        list.map((item) =>
+            cache.tryGet(String(item.link), async () => {
+                const response = await ofetch(String(item.link));
+                const $ = load(response);
+                const links = String(item.link).split('/');
+                const category = types[String(links.at(-2))];
+                item.category = [category];
+                let desc = '';
+                if (type === 'jinqigengxin') {
+                    desc = '类型：' + category + '<br>';
+                }
+                desc += '标签：' + $('.book-info__categories').first().html() + '<br>';
+                item.description = desc + $('.intro').first().html();
+                item.image = String($('.book-cover img').attr('src'));
```

Comment:
https://github.com/DIYgod/RSSHub/pull/18950#issuecomment-2833343805

Please fix the relative URL in `image`.